src/main/java/com/epam/esm/giftcertificates/validation/GiftValidate.java

```diff
@@ -0,0 +1,62 @@
+package com.epam.esm.giftcertificates.validation;
+
+import com.epam.esm.giftcertificates.entity.GiftCertificate;
+import com.epam.esm.giftcertificates.exception.CertificateInvalidRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+
+public class GiftValidate {
+    private static final String ASC = "ASC";
+    private static final String DESC = "DESC";
+
+    public static void positiveRequestedId(Long id) {
+        if (id <= 0L)
+            throw new CertificateInvalidRequest("Invalid id ( request id = " + id
+                    + " ). ID must be positive number.");
+    }
+
+    public static void findGiftByFieldsValidation(String sortByDate, String sortByName, String... ech) {
+        if (Arrays.stream(ech).allMatch(StringUtils::isEmpty) && StringUtils.isEmpty(sortByDate) && StringUtils.isEmpty(sortByName)) {
+            throw new CertificateInvalidRequest("Invalid request param. At least one param must be entered."
+            );
+        }
+        if (Arrays.stream(ech).filter(Objects::nonNull).anyMatch(String::isBlank)) {
+            throw new CertificateInvalidRequest("Invalid request param. Request param value cant be blank."
+            );
+        }
+        if (sortByDate != null && (!sortByDate.toUpperCase(Locale.ROOT).equals(ASC) && !sortByDate.toUpperCase(Locale.ROOT).equals(DESC))) {
+            throw new CertificateInvalidRequest("Invalid request param sortByDate ( request = " + sortByDate
+                    + "). Valid values (ASC,DESC)");
+        }
+        if (sortByName != null && (!sortByName.toUpperCase(Locale.ROOT).equals(ASC) && !sortByName.toUpperCase(Locale.ROOT).equals(DESC))) {
+            throw new CertificateInvalidRequest("Invalid request param sortByName ( request = " + sortByName
+                    + "). Valid values (ASC,DESC)");
+        }
+    }
+
+    public static void GiftCertificateOnUpdate(Long id, GiftCertificate updateGiftCertificate) {
+        positiveRequestedId(id);
+        certificateFieldValidation(updateGiftCertificate);
+    }
+
+    public static void certificateFieldValidation(GiftCertificate updateGiftCertificate) {
+        if (!NumberUtils.isParsable(updateGiftCertificate.getDuration())) {
+            throw new CertificateInvalidRequest("Invalid certificate field  duration ( duration = " + updateGiftCertificate.getDuration()
+                    + "). Duration must be a numeric!");
+        }
+        if (updateGiftCertificate.getPrice() != null && updateGiftCertificate.getPrice().compareTo(BigDecimal.ZERO) <= 0) {
```

Comment:
Isn't there some Apache Utils method to check if BigDecimal is negative or zero?